share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java

```diff
@@ -274,79 +270,89 @@ public CoordinatorResult<WriteShareGroupStateResponseData, CoordinatorRecord> wr
             return error.get();
         }
 
-        String groupId = request.groupId();
         WriteShareGroupStateRequestData.WriteStateData topicData = request.topics().get(0);
         WriteShareGroupStateRequestData.PartitionData partitionData = topicData.partitions().get(0);
+        SharePartitionKey key = SharePartitionKey.getInstance(request.groupId(), topicData.topicId(), partitionData.partition());
+
+        CoordinatorRecord record = getRecord(partitionData, key);
+
+        // build the response based on record
+        // errors have been already handled above
```

Comment:
Hmm, getRecord doesn't handle errors. The caller handles error from appending the record.

Reply:
Yes, the `maybeGetWriteStateError` is called before `getRecord` which does sanity and epoch checks - returns error if needed. 
The getRecord can throw a `IllegalStateException` in case it does not find valid current state.